src/coreclr/src/vm/amd64/unixstubs.cpp

```diff
@@ -10,35 +10,30 @@ extern "C"
         PORTABILITY_ASSERT("Implement for PAL");
     }
 
-    DWORD getcpuid(DWORD arg, unsigned char result[16])
+    void __cpuid(int cpuInfo[4], int function_id)
     {
-        DWORD eax;
-        __asm("  xor %%ecx, %%ecx\n" \
+        // Based on the Clang implementation provided in cpuid.h:
+        // https://github.com/llvm/llvm-project/blob/master/clang/lib/Headers/cpuid.h
+
+        __asm("  xchgq  %%rbx,%q1\n" \
               "  cpuid\n" \
-              "  mov %%eax, 0(%[result])\n" \
-              "  mov %%ebx, 4(%[result])\n" \
-              "  mov %%ecx, 8(%[result])\n" \
-              "  mov %%edx, 12(%[result])\n" \
-            : "=a"(eax) /*output in eax*/\
-            : "a"(arg), [result]"r"(result) /*inputs - arg in eax, result in any register*/\
-            : "rbx", "ecx", "edx", "memory" /* registers that are clobbered, *result is clobbered */
-          );
-        return eax;
+              "  xchgq  %%rbx,%q1" \
+            : "=a"(cpuInfo[0]), "=r"(cpuInfo[1]), "=c"(cpuInfo[2]), "=d"(cpuInfo[3]) \
+            : "0"(function_id)
+        );
     }
 
-    DWORD getextcpuid(DWORD arg1, DWORD arg2, unsigned char result[16])
+    void __cpuidex(int cpuInfo[4], int function_id, int subFunction_id)
     {
-        DWORD eax;
-        __asm("  cpuid\n" \
-              "  mov %%eax, 0(%[result])\n" \
-              "  mov %%ebx, 4(%[result])\n" \
-              "  mov %%ecx, 8(%[result])\n" \
-              "  mov %%edx, 12(%[result])\n" \
-            : "=a"(eax) /*output in eax*/\
-            : "c"(arg1), "a"(arg2), [result]"r"(result) /*inputs - arg1 in ecx, arg2 in eax, result in any register*/\
-            : "rbx", "edx", "memory" /* registers that are clobbered, *result is clobbered */
-          );
-        return eax;
+        // Based on the Clang implementation provided in cpuid.h:
+        // https://github.com/llvm/llvm-project/blob/master/clang/lib/Headers/cpuid.h
+
+        __asm("  xchgq  %%rbx,%q1\n" \
```

Comment:
Why does this need to do `xchg` instead of just telling the compiler that result is in rbx?

Reply:
(like the x86 version)

Reply:
Ah, I forgot to copy the comment from cpuid.h:
> x86-64 uses %rbx as the base register, so preserve it.

Reply:
Our original version did not have a problem with rbx. Why is it a problem for this one?

Reply:
I imagine it was done like this in cpuid.h because cpuid.h is a `#define` and so therefore it must assume preserving `rbx` is needed, while our version is a standalone function and doesn't use the stack (everything is in register already) and so doesn't need to concern itself with `rbx`.

Reply:
So delete it here since it is not needed?

Reply:
I'll remove and double check the assembly